llvm/lib/Transforms/IPO/DeadArgumentElimination.cpp

```diff
@@ -876,6 +876,8 @@ bool DeadArgumentEliminationPass::removeDeadStuffFromFunction(Function *F) {
   // it again.
   F->getParent()->getFunctionList().insert(F->getIterator(), NF);
   NF->takeName(F);
+  if (NumArgumentsEliminated)
```

Comment:
What cases are we excluding with the "if" here?  If we don't change the signature, we don't recreate the function in the first place.

Reply:
> What cases are we excluding with the "if" here? If we don't change the signature, we don't recreate the function in the first place.

IIUC, there is another metric NumRetValsEliminated. It is possible that NumRetValsEliminated > 0 and NumArgumentsEliminated = 0. I added the above check since we really care function arguments in bpf tracing and func return value is not that important. But I can remove the above check so we do not limit the case only to bpf tracing.

Reply:
I can imagine wanting a different suffix for the case where we only eliminate a return value, but I think it makes sense to have some suffix.

Reply:
Okay, I will remove the check and use '.retelim' suffix for cases where only the return value is eliminated.

Reply:
> What cases are we excluding with the "if" here? If we don't change the signature, we don't recreate the function in the first place.

Okay, uploaded new revision to introduce ".retelim" suffix is the function sole change is to remove return values.

Reply:
> I can imagine wanting a different suffix for the case where we only eliminate a return value, but I think it makes sense to have some suffix.

Just pushed a new revision to introduce '.retelim' suffix for cases where the sole func signature change is to remove return values.